typescript-test-samples/apigw-lambda-external/template.yaml

```diff
@@ -0,0 +1,150 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::Serverless-2016-10-31
+Description: >
+  API Gateway -> Lambda -> External - Serverless Testing Pattern
+
+  Sample SAM Template for API Gateway -> Lambda -> External - Serverless Testing Pattern
+
+# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
+Globals:
+  Function:
+    Timeout: 3
+    MemorySize: 128
+
+Parameters:
+  IsVolatile:
+    Description: Is this a volatile build
+    Type: String
+    Default: "false"
+
+Conditions:
+  isVolatileBuild: !Equals [!Ref IsVolatile, "true"]
+
+Resources:
+  WeatherFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: src/
+      Handler: app.lambdaHandler
+      Runtime: nodejs16.x
+      Architectures:
+        - x86_64
+      Tracing: Active
+      # Available policies:
+      # https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-policy-template-list.html
+      Policies:
+        - AWSXrayWriteOnlyAccess
+      Environment:
+        Variables:
+          WEATHER_API:
+            !If [
+              isVolatileBuild,
+              !Sub "https://${MockApi}.execute-api.${AWS::Region}.amazonaws.com/Volatile",
+              "https://goweather.herokuapp.com/weather",
+            ]
+
+      Events:
+        Weather:
+          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
+          Properties:
+            Path: /{city}
+            Method: get
+    Metadata: # Manage esbuild properties
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        # Sourcemap: true # Enabling source maps will create the required NODE_OPTIONS environment variables on your lambda function during sam build
+        EntryPoints:
+          - app.ts
+
+  # Volatile Stack Resources
+  MockApi:
```

Comment:
Should this API only be deployed in a `volatile` environment? That is, when the `IsVolatile` parameter is true?